ssd/strategy_factory.py

```diff
@@ -0,0 +1,15 @@
+from abc import ABC, abstractmethod
+from ssd.strategy import DequeStrategy, InterfaceStrategy
+
+
+class InterfaceStrategyFactory(ABC):
+    @abstractmethod
+    def select_strategy(self, strategy_name: str) -> InterfaceStrategy:
+        pass
+
+
+class StrategyFactory(InterfaceStrategyFactory):
```

Comment:
```suggestion
class StrategyFactory:
```

Factory패턴에 인터페이스는 굳이 필요하지는 않아보입니다